
libpthread is a stub library on OmniOS and does not need to be included in the
link.

diff -wpruN '--exclude=*.orig' a~/src/tcsd/Makefile.am a/src/tcsd/Makefile.am
--- a~/src/tcsd/Makefile.am	1970-01-01 00:00:00
+++ a/src/tcsd/Makefile.am	1970-01-01 00:00:00
@@ -1,7 +1,7 @@
 sbin_PROGRAMS=tcsd
 
 tcsd_CFLAGS=-DAPPID=\"TCSD\" -DVAR_PREFIX=\"@localstatedir@\" -DETC_PREFIX=\"@sysconfdir@\" -I${top_srcdir}/src/include -fPIE -DPIE
-tcsd_LDADD=${top_builddir}/src/tcs/libtcs.a ${top_builddir}/src/tddl/libtddl.a -lpthread @CRYPTOLIB@
+tcsd_LDADD=${top_builddir}/src/tcs/libtcs.a ${top_builddir}/src/tddl/libtddl.a @CRYPTOLIB@
 tcsd_LDFLAGS=@TCSD_LDFLAGS@
 tcsd_SOURCES=svrside.c tcsd_conf.c tcsd_threads.c platform.c
 
diff -wpruN '--exclude=*.orig' a~/src/tspi/Makefile.am a/src/tspi/Makefile.am
--- a~/src/tspi/Makefile.am	1970-01-01 00:00:00
+++ a/src/tspi/Makefile.am	1970-01-01 00:00:00
@@ -17,7 +17,7 @@ libtspi_la_LIBADD=${top_builddir}/src/tr
 # 5. If any interfaces have been added since the last public release, then increment age.
 # 6. If any interfaces have been removed since the last public release, then set age to 0.
 
-libtspi_la_LDFLAGS=-version-info 3:0:2 -lpthread @CRYPTOLIB@
+libtspi_la_LDFLAGS=-version-info 3:0:2 @CRYPTOLIB@
 
 libtspi_la_CFLAGS=-I$(top_srcdir)/src/include -DAPPID=\"TSPI\" -DVAR_PREFIX=\"@localstatedir@\" -DETC_PREFIX=\"@sysconfdir@\"
 
